src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -1209,7 +1208,7 @@ abstract class ClassfileParser {
       assert(runtimeAnnotStart != -1, s"No RuntimeVisibleAnnotations in classfile with ScalaSignature attribute: $clazz")
       in.bp = runtimeAnnotStart
       val numAnnots = u2
-      var i = 0
+      val i = 0
```

Comment:
Not obvious whether it's a cut/paste bug.

Reply:
This is an oversight, `i` should be incremented in the loop. https://github.com/scala/scala/pull/6816

Reply:
Reverted to var.

Reply:
You said you reverted it. Maybe you inverted it.